Dockerfile

```diff
@@ -10,109 +10,143 @@ ARG JENKINS_INBOUND_AGENT_VERSION=3309.v27b_9314fd1a_4-1
+    && rm -f "${rep_config_pkg}" \
+    && apt-get update --quiet \
+    && apt-get install --yes --no-install-recommends azcopy="${AZCOPY_VERSION}" \
+    # Sanity check
+    && azcopy --version \
+    # Cleanup
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 ## Install Azure Cli
 ARG AZ_CLI_VERSION=2.72.0
 # Download and install the Microsoft signing key
 RUN mkdir -p /etc/apt/keyrings \
-  && curl -sLS https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/keyrings/microsoft.gpg > /dev/null \
-  && chmod go+r /etc/apt/keyrings/microsoft.gpg \
-  # Add the Azure CLI software repository
-  && AZ_REPO=$(lsb_release -cs) \
-  && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/microsoft.gpg] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list \
-  # Update repository information and install the azure-cli package
-  && apt-get -y update \
-  && apt-get -y install --no-install-recommends azure-cli="${AZ_CLI_VERSION}-1~${AZ_REPO}" \
-  # Sanity check
-  && az version \
-  # Cleanup
-  && apt-get clean \
-  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
+    && curl -sLS https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor | tee /etc/apt/keyrings/microsoft.gpg > /dev/null \
+    && chmod go+r /etc/apt/keyrings/microsoft.gpg \
+    # Add the Azure CLI software repository
+    && AZ_REPO=$(lsb_release -cs) \
+    && echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/microsoft.gpg] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" | tee /etc/apt/sources.list.d/azure-cli.list \
+    # Update repository information and install the azure-cli package
+    && apt-get -y update \
+    && apt-get -y install --no-install-recommends azure-cli="${AZ_CLI_VERSION}-1~${AZ_REPO}" \
+    # Sanity check
+    && az version \
+    # Cleanup
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/*
 
 ARG TYPOS_VERSION=1.32.0
-# ARG TYPOS_SHASUM_256="27ce43632f09d5dbeb2231fe6bbd7e99eef4ed06a9149cd843d35f70a798058c"
-RUN curl --silent --show-error --location --output /tmp/typos.tar.gz \
-  "https://github.com/crate-ci/typos/releases/download/v${TYPOS_VERSION}/typos-v${TYPOS_VERSION}-x86_64-unknown-linux-musl.tar.gz" \
-  # && sha256sum /tmp/typos.tar.gz | grep -q "${TYPOS_SHASUM_256}" \
-  && tar xvfz /tmp/typos.tar.gz -C /usr/local/bin ./typos \
-  && chmod a+x /usr/local/bin/typos \
-  && typos --help
+RUN ARCH="$(dpkg --print-architecture)"; \
+    case "${ARCH}" in \
+      aarch64|arm64) \
+        DOWNLOAD_URL="https://github.com/crate-ci/typos/releases/download/v${TYPOS_VERSION}/typos-v${TYPOS_VERSION}-aarch64-unknown-linux-musl.tar.gz"; \
+        ;; \
+      amd64|x86_64) \
+        DOWNLOAD_URL="https://github.com/crate-ci/typos/releases/download/v${TYPOS_VERSION}/typos-v${TYPOS_VERSION}-x86_64-unknown-linux-musl.tar.gz"; \
+        ;; \
+      *) \
```

Comment:
uname -m instead

Reply:
I got it: 

dpkg --print-architecture : arm64/amd64
uname -m : aarch64/x86_64

right ?